.github/workflows/examples.yml

```diff
@@ -0,0 +1,55 @@
+name: Test examples
+on:
+  pull_request:
+    types: [ opened, reopened ]
+  pull_request_review:
+    types:
+      - submitted
+
+env:
+  NEST_VERSION: 3.7
+  NEST_FOLDER: ${{ github.workspace }}/.nx/installation/nest
+
+jobs:
+  test_examples:
+    if: |
+      github.event.action == 'opened' ||
+      github.event.action == 'reopened' ||
+      (github.event_name == 'pull_request_review' && github.event.review.state == 'approved')
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.list-examples.outputs.folders }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Install apt dependencies
+        run: |
+          sudo apt update
+          # Install `libopenmpi` for MPI, `libhdf5` for `morphio`,
+          # `libgsl` and `libreadline` for NEST
+          sudo apt install openmpi-bin libopenmpi-dev libhdf5-dev libreadline-dev libgsl-dev
+
+      - name: Install nodejs
+        run: |
+          curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.3/install.sh | bash
+          \. "$HOME/.nvm/nvm.sh"
+          nvm install --lts
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+
+      - name: Cache NEST
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.NEST_FOLDER }}
+          key: ${{ runner.os }}-3.12-nest
+
+      - name: Run unittests
+        run: |
+          ./nx run examples:test
+        env:
+          UV_PYTHON: 3.12
+          BSB_QUIET: true
+          NX_DAEMON: false  # may prevent GHA to stall for some reason
```

Comment:
Let's always place `NX_DAEMON: false` in the global workflow `env` block so that it's always active